tests/transition.md

```diff
@@ -277,3 +281,381 @@ All test cases described in this document are beginning in a pre-Merge world and
+    * EL's head points to `TB` (in this case `fcU` won't be applied at all because it does point to a block on an invalid chain, thus, no update and PoS transition should happen; normally, a CL will send `fcU(P1)` when get `lvh = P1.blockHash`)
+  
+  </details>
+
+* [ ] Re-org and sync with the chain having invalid *terminal* block
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- TB <- P1 <- P2`, `B: Genesis <- ...  <- INV_TB <- P1 <- P2`, `B.INV_TB` is a *valid PoW* but an *invalid terminal* block:
+    * [ ] `INV_TB.TD < TTD`
+    * [ ] `INV_TB.parent.TD >= TTD`
+  * This scenario may require second EL with a higher `TTD` value
+  * EL starts with `A.P2` as the head
+  * `newPayload(B.P2) + forkchoiceUpdated(P2)`
+    * EL returns `{status: SYNCING}`
+    * EL's head points to `A.P2`
+  * EL should pull `B: INV_TB <- P1 <- P2` from the network
+  * poll `forkchoiceUpdated(B.P2)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00.00}`
+    * EL's head points to `A.P2`
+  
+  </details>
+
+* [ ] Re-org and sync with the chain having invalid *transition* block
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- TB <- P1 <- P2`, `B: Genesis <- ... <- TB <- INV_P1 <- P2`, `B.TB` is a valid terminal block, `B.INV_P1` is an invalid payload
+  * EL starts with `A.P2` as the head
+  * `newPayload(B.P2) + forkchoiceUpdated(B.P2)`
+    * EL returns `{status: SYNCING}`
+    * EL's head points to `A.P2`
+  * poll `forkchoiceUpdated(B.P2)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00.00}`
+    * EL's head points to `A.P2`
+  
+  </details>
+
+* [ ] Re-org and sync with the chain having invalid *post-transition* block
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- TB <- P1 <- P2 <- P3`, `B: Genesis <- ... <- TB <- P1 <- INV_P2 <- P3`, `B.TB` is a valid terminal block, `B.INV_P2` is an invalid payload
+  * EL starts with `A.P2` as the head
+  * `newPayload(B.P3) + forkchoiceUpdated(B.P3)`
+    * EL returns `{status: SYNCING}`
+    * EL's head points to `A.P3`
+  * poll `forkchoiceUpdated(B.P3)`
+    * EL returns `{status: INVALID, latestValidHash: B.P1.blockHash}`
+    * EL's head points to `A.P3` (in this case `fcU` won't be applied at all because it does point to a block on an invalid chain, thus, no update of the forkchoice state should happen; normally, a CL will send `fcU(B.P1)` when get `lvh = B.P1.blockHash` and switch canonical chain to `B` with `B.P1` as the head)
+  
+  </details>
+
+* [ ] Stop processing gossiped PoW blocks
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * PoW/Clique miner builds a chain and advertises it via gossip; the chain goes beyond a terminal PoW block
+  * EL is connected to the miner and does not process blocks after the terminal one, i.e. the head points to the terminal block
+  
+  </details>
+
+* [ ] Stop processing synced PoW blocks
```

Comment:
Should be covered by https://github.com/ethereum/hive/blob/f2838a629dcd1a027a260cd10fc84899f89e3ba6/simulators/ethereum/engine/mergetests.go#L228